src/web/ng/impl/WsConnection.hpp

```diff
@@ -167,6 +166,37 @@ class WsConnection : public WsConnectionBase {
             })
         );
     }
+
+    std::optional<Error>
+    sendImpl(std::variant<Response, std::shared_ptr<std::string>> message, boost::asio::yield_context yield)
```

Comment:
This looks very similar to the http implementation. Consider adding an abstraction so that if we change one place we don't have to adjust the other each time